src/file_loader.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013-2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib::translate::*;
+use glib::object::IsA;
+use gtk_source_sys;
+use std::boxed::Box;
+use std::ptr;
+use FileLoader;
+
+impl FileLoader
+{
+    pub fn load_async<
+        P: IsA<gio::Cancellable>,
+        Q: 'static + Fn(i64, i64) + Send,
+        R: 'static + FnOnce(Result<bool, glib::Error>) + Send,
+        S: 'static + FnOnce() + Send,
+    >(
+        &self,
+        io_priority: glib::Priority,
+        cancellable: Option<&P>,
+        progress_callback: Q,
+        ready_callback: R,
+        notify_callback: S,
```

Comment:
Maybe we don't needed pass `notify_callback` to user as it internal and service callback

Reply:
My thinking was that it might be useful for user to release any resources that were acquired for progress display.

Reply:
Hm, reasonable.
Then lets leave it.

Reply:
I disagree. The `GDestroyNotify` callbacks are meant for native memory management, which in our case means to free the corresponding closures.

If any other cleanup is needed you would do this safely by passing your resources into the closures and having a `Drop` impl on their types so that you get notified when they're really not needed anymore.

Reply:
@sdroege Sounds good to me. Let's lose the user callback.